controller/network/link_controller.go

```diff
@@ -46,6 +48,22 @@ func (linkController *linkController) has(link *Link) bool {
 	return linkController.linkTable.has(link)
 }
 
+func (linkController *linkController) routerReportedLink(linkId, linkType string, src, dst *Router) (*Link, bool) {
+	linkController.lock.Lock()
+	linkController.lock.Unlock()
```

Comment:
Is locking and immediately locking here intentional? Just looked odd to me without a defer

Reply:
Good catch, thank you! Fixed.